mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -379,6 +380,17 @@ impl NexusChild {
         }
     }
 
+    /// Extract the child UUID from its name.
+    pub(crate) fn uuid(&self) -> String {
+        let url = Url::parse(&self.name).expect("Failed to parse child url");
+        for pair in url.query_pairs() {
+            if pair.0 == "uuid" {
+                return pair.1.to_string();
+            }
+        }
+        panic!("Child name does not contain a uuid query parameter.");
```

Comment:
Since the presence of UUID is perceived as a 'must have' prerequisite for NexusChild object, wouldn't it be better to add such hard check to 'new()' method to assure the presence of UUID in the name ?

Reply:
Also, we already have 'guid' member in NexusChild. Looks like it serves the same purpose as 'uuid()' member. Can we initialise this member based on URI's UUID part and then just keep using the 'guid' member ?

Reply:
Good idea. I have added a check in the `NexusChild::new` which checks for a UUID.

The `guid` member of NexusChild may be the same thing but I cannot be confident in that because the labelling code seems to conditionally overwrite it. By extracting it from the name I know that this will match up exactly with what MOAC sees.